### ISSUE TITLE
bug(gui): app icon not showing on Ubuntu

### ISSUE THREAD

**Issue by t-bre** [sufst/vcu-gui#93]
### Description
The app icon is not visible in the dock on Ubuntu (and possibly other Linux distributions?).

### Steps To Reproduce
1. Build app on Ubuntu.
2. Launch app.
3. App icon is empty.

### Environment
```text
os:        Ubuntu 20.04.5 aarch64
compiler:  gcc 9.4.0
```

### Expected Behaviour
The app icon is visible.

### Additional Notes

![Screenshot 2022-10-13 at 18 15 12 (2)](https://user-images.githubusercontent.com/31595761/195662919-ec86ace6-e70f-4edc-b8b1-e63a189ca4a0.png)

**Comment by t-bre** [comment]
@inventor02 @tgodfrey0 Does the app icon appear for you guys? Don't know what distro you're on, but I imagine it would be a general Linux issue.

**Comment by inventor02** [comment]
We dug a little deeper and found some more context as to why this happens:

On Linux for most desktop environments it is the DE itself that controls the icon of the application (so that, for example, users can theme application icons to fit in with the look of their OS). This usually (on GNOME at least) manifests itself in the form of a `.desktop` file which the user would start the application via (i.e. the `.desktop` file determines what appears in the launcher/program list).

Making a `vcu-gui.desktop` file with the following contents:

```
[Desktop Entry]
Type=Application
Icon=/home/george/sufst/vcu-gui/res/images/AppIcon_512.png
Name=VCU GUI
Terminal=false
Hidden=false
Categories=Development
Comment=The Rodin Platform is an Eclipse-based IDE for Event-B that provides effective support for refinement and mathematical proof.
Exec=/home/george/sufst/vcu-gui/build/VCU-GUI_artefacts/Debug/VCU\ GUI`
Path=/home/george/sufst/vcu-gui
```

Putting this in `~/.local/share/applications` and launching the app via the programs menu causes the icon to display as it should.

Usually this would be done by the install step of the build process or the package manager, but in development this would never happen as we run the files directly.

Basically it is not really a "bug" per se and is more a by-product of the way the desktop environment handles icons - not sure if there's a way to force this but in a production install the .desktop file would usually be created by the installer so this would not be so much of a problem.

**Comment by t-bre** [comment]
Awesome work, thanks for figuring that out. I guess if we want to be fancy and make a proper installer/package for the next _"production"_ (ha ha) release we could, but that's way down the list of priorities. Seems to be pretty easy to do and has been discussed [on the JUCE forums](https://forum.juce.com/t/standalone-app-icon-linux-makefile/44149/).

I'll leave this open for now in case we want to come back to it when we prepare the next release.
